llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -102,7 +102,8 @@ void RISCVDAGToDAGISel::PreprocessISelDAG() {
       Chain = CurDAG->getNode(ISD::TokenFactor, DL, MVT::Other, Lo, Hi);
 
       SDVTList VTs = CurDAG->getVTList({VT, MVT::Other});
-      SDValue IntID =
+      SDValue IntID = Subtarget->hasVendorXTHeadV() ?
+          CurDAG->getTargetConstant(Intrinsic::riscv_th_vlse, DL, MVT::i64) :
           CurDAG->getTargetConstant(Intrinsic::riscv_vlse, DL, MVT::i64);
```

Comment:
Yes, it is indeed a bug fix. The change mainly affects `vadd/vsub/vrsub` with GP register operands when we are using RISCV-32.

We have test cases like:

```LLVM IR
define <vscale x 1 x i64> @intrinsic_vadd_vx_nxv1i64_nxv1i64_i64(<vscale x 1 x i64> %0, i64 %1, i32 %2) nounwind {
entry:
  %a = call <vscale x 1 x i64> @llvm.riscv.th.vadd.nxv1i64.i64(
    <vscale x 1 x i64> undef,
    <vscale x 1 x i64> %0,
    i64 %1,
    i32 %2)

  ret <vscale x 1 x i64> %a
}
```

In RISCV-32, the 64-bit `%1` is cut and then passed in a0 and a1. Therefore during some point in the instruction selection stage, we have:

```
t19: nxv1i64 = RISCVISD::SPLAT_VECTOR_SPLIT_I64_VL undef:nxv1i64, t4, t6, t8
t20: nxv1i64 = llvm.riscv.th.vadd TargetConstant:i32<8857>, undef:nxv1i64, t2, t19, t8
```

The `SPLAT_VECTOR_SPLIT_I64_VL` node is generated to help handle the `%1` split across a0 and a1. It will be converted to the `vlse` intrinsic by the line above. So check whether we're using `xtheadvector` or not can avoid the backend generating instructions for RVV 1.0.

Finally we will get assembly code like this:

```assembly
sw a1, 12(sp)
sw a0, 8(sp)
addi a0, sp, 8
th.vsetvli zero, a2, e64, m1, d1
th.vlse.v v9, (a0), zero
th.vadd.vv v8, v8, v9
addi sp, sp, 16
ret
```

The intrinsic will finally turn to a `th.vlse.v` to move the 64-bit integer to the vector register.